File: FOLDERPHP/film.php
<?php
/*
Saya Nadya Rizqitha Salsabila dengan NIM 2500991 mengerjakan soal Kuis 1
dalam mata kuliah Desain Pemrograman Berorientasi Objek untuk keberkahanNya
maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.
*/

class Film
{
    private $idFilm;
    private $judul;
    private $genre;
    private $durasi;       // dalam menit
    private $hargaTiket;
    private $gambar;        // path file lokal poster film (bukan url)

    public function __construct($idFilm, $judul, $genre, $durasi, $hargaTiket, $gambar)
    {
        $this->idFilm = $idFilm;
        $this->judul = $judul;
        $this->genre = $genre;
        $this->durasi = $durasi;
        $this->hargaTiket = $hargaTiket;
        $this->gambar = $gambar;
    }

    // ===== getter =====
    public function getIdFilm() { return $this->idFilm; }
    public function getJudul() { return $this->judul; }
    public function getGenre() { return $this->genre; }
    public function getDurasi() { return $this->durasi; }
    public function getHargaTiket() { return $this->hargaTiket; }
    public function getGambar() { return $this->gambar; }

    // ===== setter =====
    public function setJudul($judul) { $this->judul = $judul; }
    public function setGenre($genre) { $this->genre = $genre; }
    public function setDurasi($durasi) { $this->durasi = $durasi; }
    public function setHargaTiket($hargaTiket) { $this->hargaTiket = $hargaTiket; }
    public function setGambar($gambar) { $this->gambar = $gambar; }
}

File: FOLDER PYTHON/Film.py
"""
Saya Nadya Rizqitha Salsabila dengan NIM 2500991 mengerjakan soal Kuis 1
dalam mata kuliah Desain Pemrograman Berorientasi Objek untuk keberkahanNya
maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.
"""


class Film:
    def __init__(self, id_film: int, judul: str, genre: str, durasi: int,
                 harga_tiket: float):
        self.id_film = id_film
        self.judul = judul
        self.genre = genre
        self.durasi = durasi          # dalam menit
        self.harga_tiket = harga_tiket

    # getter
    def get_id_film(self):
        return self.id_film
    
    # getter
    def get_judul(self):
        return self.judul

    # setter
    def set_judul(self, judul):
        self.judul = judul

    # setter
    def set_genre(self, genre):
        self.genre = genre

    # setter
    def set_durasi(self, durasi):
        self.durasi = durasi

    # setter
    def set_harga_tiket(self, harga_tiket):
        self.harga_tiket = harga_tiket

    # method untuk menampilkan info film
    def tampilkan_info(self):
        print(f"{'ID':12}: {self.id_film}")
        print(f"{'Judul':12}: {self.judul}")
        print(f"{'Genre':12}: {self.genre}")
        print(f"{'Durasi':12}: {self.durasi} menit")
        print(f"{'Harga':12}: Rp{self.harga_tiket:,.0f}")
        print("-" * 45)